### ISSUE TITLE
AdoptAtan2 getting nan after 2 steps

### ISSUE THREAD

**Issue by inspirit** [lucidrains/adam-atan2-pytorch#3]
Hey Phil! thanks for sharing another great impl!
Strangely i get nan in all losses after just 2 iterations...
I tried official implementation and it seems to work fine

**Comment by lucidrains** [comment]
@inspirit hey Eugene 😄 yes, my error, could you give it another shot? the authors also have some updates with clipping from yesterday

btw, nGPT i think they messed up their baseline, if you look at an issue in their official repo

**Comment by inspirit** [comment]
yeah! i verified that setting to zeros fix the issue :)

**Comment by lucidrains** [comment]
@inspirit let me know if you see anything interesting with this new optimizer

probably best not to get hopes up, but who knows!

**Comment by lucidrains** [comment]
@inspirit have you noticed anything? positive or negative

**Comment by inspirit** [comment]
to be honest I did not have time for deeper testing because from the start I noticed it is sensitive to learning rate and you cant simply replace existing Adam or Atan2 optimizer. it did not perform bad but it requires tuning the learning rate to squeeze good results :)

**Comment by lucidrains** [comment]
@inspirit hehe ok, did not perform bad is a pretty good sign! i also noticed it didn't perform bad
